### PR TITLE
feat: bump terra core

### PR DESCRIPTION
**Pull request by duvbell**
## Description and Motivation

* Upgrade of Migaloo's major dependencies
  * Luna
  * Alliance
  *  cosmos-sdk
  * ibc
  * comet
  * cometbft-db

## Related Issues

---
## Checklist:

- [ ] I have read [Migaloo's contribution guidelines](https://github.com/White-Whale-Defi-Platform/migaloo-chain/blob/main/docs/CONTRIBUTING.md).
- [ ] My pull request has a sound title and description (not something vague like `Update index.md`)
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation.
- [ ] The code is formatted properly `golangci-lint run ./... --fix`.

### CODE REVIEW

**Comment by faddat** [comment]
Because we are bumping other module versions, let's take this opportunity to upgrade everything else.

**Comment by faddat** [comment]
we also need to update go in the ci system

**Comment by faddat** [comment]
...and in Docker....

**Review by faddat — CHANGES_REQUESTED** [review]
We should make a new branch, release/v4.2.x and this should be tagged as v4.2.x because of breaking changes.

File: app/upgrades/v4_1_7/constants.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package v4
+
+// UpgradeName defines the on-chain upgrade name for the Migaloo v4.1.5 upgrade.
+// this upgrade includes the fix for pfm
+const (
+	UpgradeName = "v4.1.7"
```
Comment (faddat) [inline app/upgrades/v4_1_7/constants.go]: should be v4.2.0 to show breaking changes.

also, wooo 420.

**Review by faddat — APPROVED** [review]

**Review by hoank101 — APPROVED** [review]
LGTM

**Pull request merged**